packages/expo-polyfills/src/index.ts

```diff
@@ -79,3 +81,187 @@ export function polyfillMedplumWebAPIs(): void {
+    // Start clearing out each key in background
+    for (const key of this.storage.keys()) {
+      this.removeItem(key, false);
+    }
+    // Clear map
+    this.storage.clear();
+    this.setKeys();
+  }
+
+  private setKeys(): void {
+    SecureStore.setItemAsync('___keys___', JSON.stringify(Array.from(this.storage.keys()))).catch(console.error);
+  }
+
+  /**
+   * Returns the current value associated with the given key, or null if the given key does not exist.
+   * @param key - The specified storage key.
+   * @returns The current value associated with the given key, or null if the given key does not exist.
+   */
+  getItem(key: string): string | null {
+    this.assertInitialized();
+    return this.storage.get(key) ?? null;
+  }
+
+  /**
+   * Sets the value of the pair identified by key to value, creating a new key/value pair if none existed for key previously.
+   * @param key - The storage key.
+   * @param value - The new value.
+   */
+  setItem(key: string, value: string | null): void {
+    this.assertInitialized();
+    // Set key in background, or delete it if the value is null
+    if (!value) {
+      this.removeItem(key);
+    } else {
+      SecureStore.setItemAsync(key, value).catch(console.error);
+      this.storage.set(key, value);
+    }
+    this.setKeys();
+  }
+
+  /**
+   * Removes the key/value pair with the given key, if a key/value pair with the given key exists.
+   * @param key - The storage key.
+   * @param setKeys - Whether to call `this.setKeys()` after deleting the key. Defaults to `true`.
+   */
+  removeItem(key: string, setKeys = true): void {
+    this.assertInitialized();
+    SecureStore.deleteItemAsync(key).catch(console.error);
+    this.storage.delete(key);
+    if (setKeys) {
+      this.setKeys();
+    }
+  }
+
+  /**
+   * This is a stub. Returns `null`.
+   * @param _index - The numeric index. Ignored.
+   * @returns `null`.
+   */
+  key(_index: number): string | null {
+    this.assertInitialized();
+    return null;
+  }
+}
+
+interface IExpoClientStorage extends IClientStorage {
+  getInitPromise(): Promise<void>;
+  length: number;
+}
+
+/**
+ * Provides a wrapper around Expo's `SecureStore` package which provides a persistent and secure storage on mobile.
+ *
+ * This class is necessary for use with `MedplumClient` since `MedplumClient` expects `ClientStorage` to use a synchronous interface, and `SecureStore` uses an asynchronous one.
+ *
+ * On web, this class will instead wrap `localStorage` automatically, so there is no need to conditionally omit passing this to `MedplumClient` for React Native Web.
+ */
+export class ExpoClientStorage extends ClientStorage implements IExpoClientStorage {
+  // We keep a private reference to the storage we pass in so we can use it within this subclass too...
+  private secureStorage?: SyncSecureStorage;
+  constructor() {
+    // Metro should automatically prune these branches out at compile time
+    if (Platform.OS === 'web') {
```

Comment:
Does this seem like a sane way to handle the `Web` case? These checks should compile out when `Metro` compiles the final app. @codyebberson

Reply:
Seems reasonable - it's exported by Expo, so feels official.